File: laramie-comunicacion-webapi/src/domain/entities/mensaje.ts
export default class Mensaje {

    id: number;
	idTipoMensaje: number;
	idEstadoMensaje: number;
	idCanal: number;
	idPrioridad: number;
	identificador: string;
	titulo: string;
	cuerpo: string;
	idUsuarioCreacion: number;
	fechaCreacion: Date;
	fechaRecepcion: Date;
	fechaEnvio: Date;

	constructor(
        id: number = 0,
		idTipoMensaje: number = 0,
		idEstadoMensaje: number = 0,
		idCanal: number = 0,
		idPrioridad: number = 0,
		identificador: string = "",
		titulo: string = "",
		cuerpo: string = "",
		idUsuarioCreacion: number = 0,
		fechaCreacion: Date = null,
		fechaRecepcion: Date = null,
		fechaEnvio: Date = null
	)
	{
        this.id = id;
		this.idTipoMensaje = idTipoMensaje;
		this.idEstadoMensaje = idEstadoMensaje;
		this.idCanal = idCanal;
		this.idPrioridad = idPrioridad;
		this.identificador = identificador;
		this.titulo = titulo;
		this.cuerpo = cuerpo;
		this.idUsuarioCreacion = idUsuarioCreacion;
		this.fechaCreacion = fechaCreacion;
		this.fechaRecepcion = fechaRecepcion;
		this.fechaEnvio = fechaEnvio;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.idTipoMensaje = row.idTipoMensaje ?? 0;
		this.idEstadoMensaje = row.idEstadoMensaje ?? 0;
		this.idCanal = row.idCanal ?? 0;
		this.idPrioridad = row.idPrioridad ?? 0;
		this.identificador = row.identificador ?? "";
		this.titulo = row.titulo ?? "";
		this.cuerpo = row.cuerpo ?? "";
		this.idUsuarioCreacion = row.idUsuarioCreacion ?? 0;
		this.fechaCreacion = row.fechaCreacion ?? null;
		this.fechaRecepcion = row.fechaRecepcion ?? null;
		this.fechaEnvio = row.fechaEnvio ?? null;
	}

}

File: modules/comunicacion/models/mensaje.py
from sqlalchemy import Column, BigInteger, String, DateTime, func

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Mensaje(Base):
    __tablename__ = "comunicacion_mensajes"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    id_tipo_mensaje = Column(BigInteger, nullable=False)
    id_estado_mensaje = Column(BigInteger, nullable=False, default=10)
    id_canal = Column(BigInteger, nullable=False)
    id_prioridad = Column(BigInteger, nullable=False, default=10)
    identificador = Column(String(250), nullable=False)
    titulo = Column(String(250), nullable=False)
    cuerpo = Column(String(1000), nullable=False)
    id_usuario_creacion = Column(BigInteger, nullable=True)
    fecha_creacion = Column(DateTime, server_default=func.now(), nullable=False)
    fecha_recepcion = Column(DateTime, nullable=True)
    fecha_envio = Column(DateTime, nullable=True)
